File: platform/fabric/src/main/java/dev/obscuria/fragmentum/fabric/FabricFragmentumClient.java
package dev.obscuria.fragmentum.fabric;

import dev.obscuria.fragmentum.api.v1.client.FragmentumClientEvents;
import dev.obscuria.fragmentum.core.CoreFragmentumClient;
import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;

public final class FabricFragmentumClient implements ClientModInitializer
{
    @Override
    public void onInitializeClient()
    {
        CoreFragmentumClient.init();

        WorldRenderEvents.START.register(context -> FragmentumClientEvents.START_RENDER
                .broadcast(FragmentumClientEvents.WorldRender::invoke));
        WorldRenderEvents.END.register(context -> FragmentumClientEvents.END_RENDER
                .broadcast(FragmentumClientEvents.WorldRender::invoke));
        ClientTickEvents.START_WORLD_TICK.register(level -> FragmentumClientEvents.START_WORLD_TICK
                .broadcast(listener -> listener.invoke(level)));
        ClientTickEvents.END_WORLD_TICK.register(level -> FragmentumClientEvents.END_WORLD_TICK
                .broadcast(listener -> listener.invoke(level)));
        ClientTickEvents.START_CLIENT_TICK.register(minecraft -> FragmentumClientEvents.START_CLIENT_TICK
                .broadcast(listener -> listener.invoke(minecraft)));
        ClientTickEvents.END_CLIENT_TICK.register(minecraft -> FragmentumClientEvents.END_CLIENT_TICK
                .broadcast(listener -> listener.invoke(minecraft)));
    }
}

File: fabric/src/main/kotlin/dev/obscuria/fragmentum/fabric/FabricFragmentumClient.kt
package dev.obscuria.fragmentum.fabric

import dev.obscuria.fragmentum.FragmentumClient
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents.*
import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents.END
import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents.START
import dev.obscuria.fragmentum.client.FragmentumClientEvents as Client

fun clientInit() {

    FragmentumClient.init()

    START.register { Client.START_RENDER.broadcast { it.invoke() } }
    END.register { Client.END_RENDER.broadcast { it.invoke() } }
    START_WORLD_TICK.register { l -> Client.START_WORLD_TICK.broadcast { it.invoke(l) } }
    END_WORLD_TICK.register { l -> Client.END_WORLD_TICK.broadcast { it.invoke(l) } }
    START_CLIENT_TICK.register { c -> Client.START_CLIENT_TICK.broadcast { it.invoke(c) } }
    END_CLIENT_TICK.register { c -> Client.END_CLIENT_TICK.broadcast { it.invoke(c) } }
}